### PR TITLE
add some z3 packages

### PR DESCRIPTION
**Pull request by c-cube**
another attempt, which should work better (let's try it!)

note: 4.5.1 is a weird release but we need it at AestheticIntegration for some reasons.

### CODE REVIEW

**Comment by c-cube** [comment]
So, it seems opam<2 doesn't know how to fetch a package from a git repo + commit hash? Or is that something else?

**Comment by jpdeplaix** [comment]
I suppose it's because opam1 sees it as a http link where it should be advertised as a git link. Have you tried to use ```git``` instead of ```src``` in the url file ?

But anyway, I would feel kind of bad by merging a package version that seems not advertised officially. Is that really necessary for opam users ? Is your company going to add packages relying on this ? Otherwise it seems wiser to just take the 3.4.6 and that you take 3.4.5.1 on to private repo for you use cases.

**Comment by c-cube** [comment]
Yeah, that sounds really reasonable. I started from an internal package but am just going to put 4.6.0 for now.

File: packages/z3/z3.4.6/opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  [ "python2.7" "scripts/mk_make.py" "--ml" ]
+  [ make "-C" "build"]
+  [ "sh" "-c" "cp build/libz3* build/api/ml/" ]
+]
+install: [
+  [ "sh" "-c" "ocamlfind install z3 build/api/ml/META -nodll build/api/ml/*" ]
+]
+remove: ["ocamlfind" "remove" "z3"]
+depends: [
+  "ocamlfind"
```
Comment (jpdeplaix) [inline packages/z3/z3.4.6/opam]: Do you really need the ```Findlib``` library ?

**Comment by c-cube** [comment]
For some reason, fedora fails to install python2.7…

**Comment by jpdeplaix** [comment]
That's because there is no system packages defined for fedora in ```conf-python-2-7```. I'll make the necessary adjustments later. Good to merge for now. Thanks !

**Pull request merged**